qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -236,6 +239,39 @@ class AnalysisExample(CurveAnalysis):
     #: List[str]: Fixed parameter in fit function. Value should be set to the analysis options.
     __fixed_parameters__ = list()
 
+    # Automatically generated fitting functions of child class
+    _fit_model = None
+
+    def __init_subclass__(cls, **kwargs):
+        """Parse series definition of subclass and set fit function and signature."""
+
+        super().__init_subclass__(**kwargs)
```

Comment:
This is not good direction in several reasons. The primary reason is simply the performance. Inspecting a callable (note that `inspect` is not lightweight computation) every time the class is instantiated will induce significant computation overhead (current code is kind of doing this) though this doesn't change from instance to instance. For example, if we run parallel experiment on some 100Q device, I guess this overhead cannot be ignored. Another reason is parsed function signature is used to generate a template of `p0` and `bounds` in the default options, which is class method thus the signature should be parsed before the class is instantiated.

For DRAG and Ham tomo, dynamic generation of fit model doesn't help because we also need filter kwargs. I don't want to tie filter kwargs to the fit model since this will induce unnecessary coupling between fitting and data processing, which we are trying to separate.

Reply:
These considerations must be documented somewhere (in a visible place) in the code. Can be at the beginning of `__init_subclass__`. Otherwise someone will come and say "hey, it's not supposed to be a class method, let's set this up at init time`, without being aware of the potential effect on large parallel experiments.

Reply:
Good point :)